rsmtool/notebooks/header.ipynb

```diff
@@ -152,6 +154,16 @@
     "javascript_path = os.environ.get(\"JAVASCRIPT_PATH\")"
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# initialize thumbnail converter object\n",
+    "thumbnail = Thumbnail()"
```

Comment:
Ah, I think I have misunderstood how you use `Thumbnail`. You are essentially using it as a singleton just instantiating it once and then using it to convert any new file. So, an instance of `Thumbnail` is not really per image file. It's just a converter that keeps track of IDs - that's a little strange way to do it but I guess it's more efficient than having multiple Thumbnail instances around. However, in this case, we need to call it something else, perhaps `ThumbnailConverter` and make it clear how we use it.

Reply:
Sure, I'll rename it and address the comments below. My logic for using this pattern was twofold:

(1) I think it's more efficient, as you said, and (2) it ensures that `ids` are tracked and incremented, and therefore always unique.